core/src/main/java/org/axonframework/queryhandling/QueryGateway.java

```diff
@@ -157,6 +157,48 @@
                                  updateResponseType);
     }
 
+    /**
+     * Sends given {@code query} over the {@link QueryBus} and returns result containing initial response and
+     * incremental updates (received at the moment the query is sent, until it is cancelled by the caller or closed by
+     * the emitting side).
+     *
+     * Expects initial and incremental responses to have the same responseType.
+     *
+     * @param query               The {@code query} to be sent
+     * @param responseType The response type used for this query
+     * @param <Q>                 The type of the query
+     * @param <R>                 The type of the initial and incremental response
+     * @return registration which can be used to cancel receiving updates
+     * @see QueryGateway#subscriptionQuery(String, Object, Class)
+     */
+    default <Q, R> SubscriptionQueryResult<R, R> subscriptionQuery(Q query, Class<R> responseType) {
+        return subscriptionQuery(query.getClass().getName(),
+                                 query,
+                                 responseType);
+    }
+
+    /**
+     * Sends given {@code query} over the {@link QueryBus} and returns result containing initial response and
+     * incremental updates (received at the moment the query is sent, until it is cancelled by the caller or closed by
+     * the emitting side).
+     *
+     * Expects initial and incremental responses to have the same responseType.
+     *
+     * @param queryName           A {@link String} describing query to be executed
+     * @param query               The {@code query} to be sent
+     * @param responseType The response type used for this query
+     * @param <Q>                 The type of the query
+     * @param <R>                 The type of the initial and incremental response
+     * @return registration which can be used to cancel receiving updates
+     * @see QueryGateway#subscriptionQuery(Object, Class, Class)
+     */
+    default <Q, R> SubscriptionQueryResult<R, R> subscriptionQuery(String queryName, Q query, Class<R> responseType) {
+        return subscriptionQuery(queryName,
+                                 query,
+                                 responseType,
+                                 responseType);
+    }
+
```

Comment:
I'm missing two methods here: 
`<Q, R> SubscriptionQueryResult<R, R> subscriptionQuery(Q query, ResponseType<R> responseType)`
and
`<Q, R> SubscriptionQueryResult<R, R> subscriptionQuery(String queryName, ResponseType<R> responseType)`

Reply:
wouldn't be some kind of subscription builder be the better approach given the variety of possible parameter options?

Reply:
Although this is different from how we've set up the command/event bus api's, I think a `SubscriptionQueryBuilder` does deserve some thinking. Let's discuss this @abuijze / @m1l4n54v1c

Reply:
If we're not taking that route, then I do agree with @m1l4n54v1c that we should have the other two API calls he suggests.

Reply:
Sure, why not, @jangalinski could you create a proposal for it?

Reply:
Sure, will do

Reply:
I think this is a definitely a way to go. Unfortunately, this issue is tied to a minor framework release and proposed changes will break our backward compatibility. I would go for now with the addition of those two methods to the `QueryGateway` API. We could extract this proposal into a separate issue for a major release. Opinions?

Reply:
Thanks for being open to the change ... If there is a chance to switch to fluent builders for subscriptions, then maybe we do not bloat the existing api any more by adding convenience methods ...

And backward compatible: I let the old subcriptionQuery()  use the new builder, so internally everything is still compatible ... or did I get you wrong.

Plus: "major" means: possibly with axon 4? or is "major" the 3.4?

Reply:
Major version is 4.0.0. Your proposal is to deprecate the "convenience methods", introduce the fluent builder and rewire the convenience methods to use the builder ones? That could work, I'm just wondering whether we want to deprecate just released methods. @abuijze / @smcvb ?